apps/astarte_appengine_api/lib/astarte_appengine_api_web/router.ex

```diff
@@ -19,106 +19,171 @@ defmodule Astarte.AppEngine.APIWeb.Router do
+
+  pipeline :interface_values_device_group_api do
+    plug Astarte.AppEngine.APIWeb.Plug.JoinPath
+    plug Astarte.AppEngine.APIWeb.Plug.LogInterface
+    plug Astarte.AppEngine.APIWeb.Plug.LogPath
+  end
+
+  pipeline :device_group_name_api do
+    plug Astarte.AppEngine.APIWeb.Plug.LogDeviceId
+    plug Astarte.AppEngine.APIWeb.Plug.LogGroupName
+  end
+
   pipeline :swagger do
     plug :maybe_halt_swagger
   end
 
   scope "/v1/:realm_name", Astarte.AppEngine.APIWeb do
-    pipe_through :api
+    pipe_through :auth_api
 
     get "/version", VersionController, :show
 
     scope "/stats" do
       get "/devices", StatsController, :show_devices_stats
     end
 
-    resources "/devices", DeviceStatusController,
-      only: [:index, :show, :update],
-      param: "device_id"
+    scope "/devices" do
+      pipe_through :device_api
+
+      resources "/", DeviceStatusController,
+        only: [:index, :show, :update],
+        param: "device_id"
 
-    resources "/devices-by-alias", DeviceStatusByAliasController,
-      only: [:index, :show, :update],
-      param: "device_alias"
+      scope "/:device_id" do
+        pipe_through :interface_value_api
 
-    resources "/devices/:device_id/interfaces", InterfaceValuesController,
-      only: [:index, :show],
-      param: "interface"
+        resources "/interfaces", InterfaceValuesController,
+          only: [:index, :show],
+          param: "interface"
 
-    get "/devices/:device_id/interfaces/:interface/*path_tokens", InterfaceValuesController, :show
+        get "/interfaces/:interface/*path_tokens", InterfaceValuesController, :show
 
-    put "/devices/:device_id/interfaces/:interface/*path_tokens",
-        InterfaceValuesController,
-        :update
+        put "/interfaces/:interface/*path_tokens",
+            InterfaceValuesController,
+            :update
 
-    post "/devices/:device_id/interfaces/:interface/*path_tokens",
-         InterfaceValuesController,
-         :update
+        post "/interfaces/:interface/*path_tokens",
+             InterfaceValuesController,
+             :update
+
+        delete "/interfaces/:interface/*path_tokens",
+               InterfaceValuesController,
+               :delete
+      end
+    end
 
-    delete "/devices/:device_id/interfaces/:interface/*path_tokens",
-           InterfaceValuesController,
-           :delete
+    scope "/devices-by-alias" do
+      pipe_through :device_alias_api
 
-    resources "/devices-by-alias/:device_alias/interfaces",
-              InterfaceValuesByDeviceAliasController,
-              only: [:index, :show],
-              param: "interface"
+      resources "/", DeviceStatusByAliasController,
+        only: [:index, :show, :update],
+        param: "device_alias"
 
-    get "/devices-by-alias/:device_alias/interfaces/:interface/*path_tokens",
-        InterfaceValuesByDeviceAliasController,
-        :show
+      scope "/:device_alias" do
+        pipe_through :interface_values_device_group_api
```

Comment:
This scope could be used for more than just interfaces, so it's odd to enforce a pipeline on interface values.

We could use a scope directly on interfaces instead:
```suggestion
      scope "/:device_alias/interfaces" do
        pipe_through :interface_values_api
        
        resources "/", InterfaceValuesByDeviceAliasController,,
          only: [:index, :show],
          param: "interface"
        
        ...
```